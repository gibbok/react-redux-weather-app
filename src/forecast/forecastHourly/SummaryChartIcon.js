import React, {Component} from 'react'
import IconWeather from '../../shared/icon/IconWeather'

class SummaryChartIcon extends Component {
  render () {
    const { cx, cy, payload: {weatherIconCode} } = this.props
    return (
      <svg x={cx - 10} y={cy + 20}>
        <foreignObject width='100%' height='100%'>
          <IconWeather code={weatherIconCode} />
        </foreignObject>
      </svg>
    )
  }
}
export default SummaryChartIcon
