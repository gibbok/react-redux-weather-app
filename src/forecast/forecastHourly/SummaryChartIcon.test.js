// @flow
import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import SummaryChartIcon from './SummaryChartIcon'

describe('<SummaryChartIcon />', () => {
  it('should render', () => {
    const wrapper = shallow(
      <SummaryChartIcon
        payload={{weatherIconCode: 200}} />
      )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
