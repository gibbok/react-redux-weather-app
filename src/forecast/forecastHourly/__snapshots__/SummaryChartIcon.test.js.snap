// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryChartIcon /> should render 1`] = `
<svg
  x={NaN}
  y={NaN}
>
  <foreignObject
    height="100%"
    width="100%"
  >
    <IconWeather
      code={200}
    />
  </foreignObject>
</svg>
`;
